distributed/client.py

```diff
@@ -4053,8 +4053,64 @@ class will be instantiated with any extra keyword arguments.
         if isinstance(plugin, type):
             plugin = plugin(**kwargs)
 
+        if name is None:
+            name = _get_worker_plugin_name(plugin)
+
+        assert name
+
         return self.sync(self._register_worker_plugin, plugin=plugin, name=name)
 
+    async def _unregister_worker_plugin(self, name):
+        responses = await self.scheduler.unregister_worker_plugin(name=name)
+
+        for response in responses.values():
+            if response["status"] == "error":
+                exc = response["exception"]
+                tb = response["traceback"]
+                raise exc.with_traceback(tb)
+        return responses
+
+    def unregister_worker_plugin(self, name):
+        """Unregisters a lifecycle worker plugin
+
+        This unregister a worker plugin using its name attribute by calling
+        the plugin's teardown method. (See the ``dask.distributed.WorkerPlugin``
+        class or the examples below for the interface and docstrings.)
+
+        Parameters
+        ----------
+        name: str
+            Name of the plugin as it was registered.
+            See ``register_worker_plugin`` docstrings for more reference.
+            If there was not a name assigned when the plugin was added, it will
+            automatically assign ``name = funcname(plugin) + "-" + str(uuid.uuid4())``
```

Comment:
We can link directly to the `register_worker_plugin` API docs by using `:meth:`

```suggestion
            Name of the plugin to unregister. See the :meth:`Client.register_worker_plugin`
            docstring for more information.
```

Additionally, I think the part about the random name generation should be moved to the `register_worker_plugin` docstring since that's where that logic is actually executed. Maybe something like "If not provided, a random name will be generated".

Reply:
Makes sense, I added the random name generation part to the register_worker_plugin docstring